src/session.rs

```diff
@@ -0,0 +1,168 @@
+            xh: env!("CARGO_PKG_VERSION").into(),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Cookie {
+    name: String,
+    value: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    expires: Option<i64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    path: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    secure: Option<bool>,
+}
+
+impl Cookie {
+    fn has_expired(&self) -> bool {
+        let start = SystemTime::now();
+        let since_the_epoch = start.duration_since(UNIX_EPOCH).unwrap();
+        match self.expires {
+            Some(expires) => u128::try_from(expires).unwrap() > since_the_epoch.as_millis(),
+            None => false,
+        }
+    }
+}
+
+impl FromStr for Cookie {
+    type Err = cookie::ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let c = cookie::Cookie::parse(s)?;
+        Ok(Cookie {
+            name: c.name().into(),
+            value: c.value().into(),
+            expires: c
+                .expires()
+                .and_then(|v| v.datetime())
+                .map(|v| v.unix_timestamp()),
+            path: c.path().map(Into::into),
+            secure: c.secure(),
+        })
+    }
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+struct Content {
+    #[serde(rename = "__meta__")]
+    meta: Meta,
+    cookies: HashMap<String, Cookie>,
+    headers: HashMap<String, String>,
+}
+
+pub struct Session {
+    pub path: PathBuf,
+    pub read_only: bool,
+    content: Content,
+}
+
+impl Session {
+    pub fn load_session(host: &str, name_or_path: &str, read_only: bool) -> Result<Self> {
+        let path = if name_or_path.contains(std::path::is_separator) {
+            PathBuf::from(name_or_path)
+        } else {
+            let mut path = dirs::config_dir()
+                .context("couldn't get config directory")?
+                .join::<PathBuf>(["xh", "sessions", host].iter().collect());
+            path.push(format!("{}.json", name_or_path));
+            path
+        };
+
+        let content = match fs::read_to_string(&path) {
+            Ok(content) => serde_json::from_str::<Content>(&content)?,
+            Err(err) if err.kind() == io::ErrorKind::NotFound => Content::default(),
+            Err(err) => return Err(err.into()),
+        };
+
+        Ok(Session {
+            path,
+            read_only,
+            content,
+        })
+    }
+
+    pub fn headers(&self) -> Result<HeaderMap> {
+        let mut headers: HeaderMap = HeaderMap::try_from(&self.content.headers)?;
+        let cookies = self
+            .content
+            .cookies
+            .values()
+            .filter(|c| c.has_expired())
+            .map(|c| format!("{}={}", c.name, c.value))
+            .collect::<Vec<_>>()
+            .join("; ");
+        if !cookies.is_empty() {
+            headers.insert(COOKIE, HeaderValue::from_str(&cookies)?);
+        }
```

Comment:
I think this is too simple. HTTPie merges cookies, so if your session file has a `foo=bar` cookie and you have `cookie:baz=qux` as a request item then it sends `Cookie: foo=bar; baz=qux`.